src/google/cloud/ndb/query.py

```diff
@@ -51,10 +53,46 @@
 
 
 class QueryOptions:
-    __slots__ = ()
+    __slots__ = (
+        "client",
+        "kind",
+        "project",
+        "namespace",
+        "ancestor",
+        "filters",
+        "projection",
+        "order",
+        "distinct_on",
+        "limit",
+        "offset",
+        "start_cursor",
+        "end_cursor",
+        "eventual",
+    )
+
+    def __init__(self, config=None, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+        if config is not None:
```

Comment:
Seems like you could just copy the values from `config` first, and then let values from `kwargs` overwrite. That would simply the logic here a little bit.

Reply:
Thanks, will do.